src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py

```diff
@@ -270,6 +270,112 @@ def test_aks_byo_appgw_with_ingress_appgw_addon(self, resource_group, resource_g
             'addon_client_id': addon_client_id,
         })
 
+    @AllowLargeResponse()
+    @ResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_create_with_openservicemesh_addon(self, resource_group, resource_group_location):
+        aks_name = self.create_random_name('cliakstest', 16)
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': aks_name,
+        })
+
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} --enable-managed-identity --service-principal xxxx --client-secret yyyy --generate-ssh-keys ' \
+                     '-a open-service-mesh -o json'
+        self.cmd(create_cmd, checks=[
+            self.check('provisioningState', 'Succeeded'),
+            self.check('addonProfiles.openServiceMesh.enabled', True),
+        ])
+
+    @AllowLargeResponse()
+    @ResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_create_with_openservicemesh_addon_with_mesh_name(self, resource_group, resource_group_location):
+        aks_name = self.create_random_name('cliakstest', 16)
+        osm_mesh_name = self.create_random_name('cliakstest', 16)
```

Comment:
nit: name another osm_mesh_name? cliakstest seems not related to the mesh name.

Reply:
I see what you mean, but any string that meets k8s label value requirements works fine here. I'll consider chating this to ue a deterministic name in a follow-up.